src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -44,10 +44,7 @@ public bool Succeeded
         {
             get
             {
-                if (!_bestResultState.HasValue())
-                {
-                    _bestResultState = TryGetBestResult(this.ResultsBuilder, out _bestResult);
-                }
+                EnsureBestResultLoaded();
```

Comment:
I don't think there's any impact. All of the methods which call `EnsureBestResultLoaded` implicitly depended on the best result having been loaded (i.e., they required _bestResultState to not be Unknown) and sometimes `Assert`ed this fact. Previously, there was simply an implicit requirement that `Succeeded` was called before any method that accessed the best result, so the cost for loading would have been paid there. Now, Succeeded does not have to be called, but the cost will be paid as late as possible since only functions which actually use a non-unknown best result have `EnsureBestResultLoaded` calls.

---
In reply to: [61031785](https://github.com/dotnet/roslyn/pull/10826#discussion_r61031785) [](ancestors = 61031785)